front/src/plugins/kakao-map.js

```diff
@@ -241,6 +260,67 @@ export const KakaoMap = (() => {
     postOverlays = [];
   };
 
+  const initPlaceSearch = () => {
+    places = _createPlaces();
+    placeOverlay = _createPlaceOverlay();
+  };
+
+  const searchPlace = (place, reject) => {
+    clearPlaceMarkers();
+
+    places.keywordSearch(place, (data, status) =>
+      _searchPlaceCallBack(data, status, reject),
+    );
+  };
+
+  const _searchPlaceCallBack = (data, status, reject) => {
+    if (status === kakao.maps.services.Status.ZERO_RESULT) {
+      const error = new Error(
+        kakao.maps.services.Status.ZERO_RESULT + " - there's no result",
+      );
+      reject(error);
+    }
+    if (status === kakao.maps.services.Status.OK) {
+      const bounds = new kakao.maps.LatLngBounds();
+
+      for (let i = 0; i < data.length; i++) {
+        _displayMarker(data[i]);
+        bounds.extend(new kakao.maps.LatLng(data[i].y, data[i].x));
+      }
+
+      map.setBounds(bounds);
+    }
+  };
+
+  const _displayMarker = (place) => {
+    const marker = new kakao.maps.Marker({
+      map: map,
+      position: new kakao.maps.LatLng(place.y, place.x),
+    });
+
+    placeMarkers.push(marker);
+
+    kakao.maps.event.addListener(marker, EVENT_TYPE.CLICK, function () {
+      placeOverlay.setContent(
+        "<div class='place-overlay-style font-size-x-small'>" +
+          place.place_name +
+          "</div>",
+      );
+      placeOverlay.setPosition(new kakao.maps.LatLng(place.y, place.x));
+      placeOverlay.setMap(map);
```

Comment:
```suggestion
    if (!placeOverlay.getMap()) {
      placeOverlay.setContent(
        `<div class='place-overlay-style font-size-x-small'>${name}</div>`,
      );
      placeOverlay.setPosition(new kakao.maps.LatLng(place.y, place.x));
      placeOverlay.setMap(map);
    } else {
      placeOverlay.setMap(null);
    }
```

다시 눌렀을 때는 꺼지도록 하면 어떨까요?